### PR TITLE
update: Los Santos Lottery

### LINKED ISSUE #1426: lotto tickets
players can buy lotto tickets and it will randomly pick a winner?

like once a month or every 2 weeks smth?

good way for people to spend their money and have a chance to win smth

can also keep a portion of the money earned to go into government funds

### PR DESCRIPTION
**Pull request by Savage912-Official**
This will allow players to buy a lottery ticket for $500 and every 14 days a winner will be auto drawn. Once the drawing has happened a chat message will be put out with the winning ticket number.

At life invader peak the desk or ped to access to lottery menu for buying ticket & claiming prize.

A player can only buy 1 ticket every 14 days. You buy 1 then once the lottery resets (After a player claims their winnings) you can then buy another for an entry in a new lottery.

Closes #1426

### CODE REVIEW

File: resources/usa_lottery/client/cl_lottery.lua [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+function loadAnimDict(dict)
+    while not HasAnimDictLoaded(dict) do
+        RequestAnimDict(dict)
+        Wait(5)
+    end
+end
+
+Citizen.CreateThread(function()
+    exports['qb-target']:AddCircleZone("lotterymenuspot", Config.TargetLocation, 2.0, {
```
Comment (minipunch) [inline resources/usa_lottery/client/cl_lottery.lua:27]: We have qb-target? R u sure this actually works lol

File: resources/usa_lottery/shared/sh_config.lua [hunk 2]
```diff
@@ -0,0 +1,11 @@
+Config = {}
+
+Config.PedModel = "a_m_y_smartcaspat_01" -- Ped model you wish to use.
+
+Config.PedLocation = vector4(-1083.13, -245.57, 36.76, 200.8) -- This is where the ped will spawn.
+
+Config.TicketPrice = 500 -- Total amount the tickets cost.
```
Comment (minipunch) [inline resources/usa_lottery/shared/sh_config.lua:7]: We should raise this a bit to like 2000 at least I'd say, maybe even 5k

**Review by minipunch — CHANGES_REQUESTED** [review]

File: resources/usa_lottery/server/sv_lottery.lua [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+end)
+
+RegisterNetEvent("usa_lottery:purchaselottery", function()
+    local src = source
+    local char = exports["usa-characters"]:GetCharacter(src)
+    local cid = char.get("_id")
+    local cost = Config.TicketPrice
+    local ticketnumber = math.random(1, 500)
+	
+    if char.get("money") >= cost then
```
Comment (minipunch) [inline resources/usa_lottery/server/sv_lottery.lua:84]: Can you use `char.hasEnoughMoney` or whatever it is instead

That way we can use `char.removeMoneyOrBank(amount)` so it takes from bank or cash on hand

File: resources/usa_lottery/server/sv_lottery.lua [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+        char.removeMoney(cost)
+        TriggerClientEvent('usa:notify', src, 'You have bought a lottery ticket!')
+        MySQL.insert("INSERT INTO `lottery` (`cid`, `ticketnumber`) VALUES ('"..cid.."', '"..ticketnumber.."')")
+        local check = 'placeholder'
+        local totalamount = MySQL.query.await('SELECT * FROM lotterytotal where lotto = ?', {check})
+        local newamount = totalamount[1].total + cost
+        MySQL.update("UPDATE lotterytotal SET total = '"..newamount.."' WHERE lotto = '"..check.."'")     
+        char.giveItem(Item)
+    elseif char.get("bank") then
```
Comment (minipunch) [inline resources/usa_lottery/server/sv_lottery.lua:93]: Then you also don't need a separate `if` statement of duplicated code if use that instead

File: resources/usa_lottery/client/cl_lottery.lua (hunk 1 above)
Comment (Savage912-Official) [reply resources/usa_lottery/client/cl_lottery.lua:27]: No but our target allows for these exports and it works.

File: resources/usa_lottery/shared/sh_config.lua (hunk 2 above)
Comment (Savage912-Official) [reply resources/usa_lottery/shared/sh_config.lua]: Done

File: resources/usa_lottery/server/sv_lottery.lua (hunk 4 above)
Comment (Savage912-Official) [reply resources/usa_lottery/server/sv_lottery.lua]: Rone

File: resources/usa_lottery/server/sv_lottery.lua (hunk 3 above)
Comment (Savage912-Official) [reply resources/usa_lottery/server/sv_lottery.lua]: Done

**Review by minipunch — COMMENTED** [review]
Did I miss the part where it gives some % of the lottery total to the gov jobs ?

**Comment by minipunch** [comment]
Also in the future please include the fact that you have to run the .sql file if it includes one plz

**Comment by Savage912-Official** [comment]
> Also in the future please include the fact that you have to run the .sql file if it includes one plz
> 
> Also this is a nice thing about our setup with CouchDB rn -- you don't have to do any manual DB setup steps since it can be automated
> 
> Maybe next time u can try converting it to no sql lol
> 
> Idk why that isn't a thing for sql scripts lol

Yeah I’ll give it a go just not the best with couchdb

**Comment by Savage912-Official** [comment]
> Also in the future please include the fact that you have to run the .sql file if it includes one plz
> 
> Also this is a nice thing about our setup with CouchDB rn -- you don't have to do any manual DB setup steps since it can be automated
> 
> Maybe next time u can try converting it to no sql lol
> 
> Idk why that isn't a thing for sql scripts lol

And yeah I didn’t think about saying anything about running the sql since I had said it in dms

**Comment by minipunch** [comment]
Did you actually test this to make sure the winning worked?

**Review by minipunch — APPROVED** [review]

**Pull request merged**